Sources/AsyncKit/ConnectionPool/EventLoopConnectionPool.swift

```diff
@@ -188,9 +188,10 @@ public final class EventLoopConnectionPool<Source> where Source: ConnectionPoolS
         }
 
         // Find an available connection that isn't closed
-        while let conn = self.available.popLast() {
+        while var conn = self.available.popLast() {
             if !conn.isClosed {
                 logger.trace("Using available connection")
+                conn.lastUsed = Date()
```

Comment:
Since `conn` is now not a reference type, this won't work. You can mutate it, but the change is not stored

Reply:
yeah but the element is `popLast` out of the deque, so there is nothing stored anymore to be mutated at all.